includes/mutation/class-order-update.php

```diff
@@ -0,0 +1,149 @@
+
+/**
+ * Class Order_Update
+ */
+class Order_Update {
+	/**
+	 * Registers mutation
+	 */
+	public static function register_mutation() {
+		register_graphql_mutation(
+			'updateOrder',
+			array(
+				'inputFields'         => self::get_input_fields(),
+				'outputFields'        => self::get_output_fields(),
+				'mutateAndGetPayload' => self::mutate_and_get_payload(),
+			)
+		);
+	}
+
+	/**
+	 * Defines the mutation input field configuration
+	 *
+	 * @return array
+	 */
+	public static function get_input_fields() {
+		$input_fields = array_merge(
+			Order_Create::get_input_fields(),
+			array(
+				'id'         => array(
+					'type'        => 'ID',
+					'description' => __( 'Order global ID', 'wp-graphql-woocommerce' ),
+				),
+				'order'      => array(
+					'type'        => 'Int',
+					'description' => __( 'Order WP ID', 'wp-graphql-woocommerce' ),
+				),
+				'customerId' => array(
+					'type'        => 'Int',
+					'description' => __( 'Order customer ID', 'wp-graphql-woocommerce' ),
+				),
+			)
+		);
+
+		return $input_fields;
+	}
+
+	/**
+	 * Defines the mutation output field configuration
+	 *
+	 * @return array
+	 */
+	public static function get_output_fields() {
+		return array(
+			'order' => array(
+				'type'    => 'Order',
+				'resolve' => function( $payload ) {
+					return new Order( $payload['id'] );
+				},
+			),
+		);
+	}
+
+	/**
+	 * Defines the mutation data modification closure.
+	 *
+	 * @return callable
+	 */
+	public static function mutate_and_get_payload() {
+		return function( $input, AppContext $context, ResolveInfo $info ) {
+			$post_type_object = get_post_type_object( 'shop_order' );
+
+			if ( ! current_user_can( $post_type_object->cap->create_posts ) ) {
+				throw new UserError( __( 'Sorry, you are not allowed to update this order.', 'wp-graphql-woocommerce' ) );
+			}
+
+			// Retrieve order ID.
+			$order_id = null;
+			if ( ! empty( $input['id'] ) ) {
+				$id_components = Relay::fromGlobalId( $input['id'] );
+				if ( empty( $id_components['id'] ) || empty( $id_components['type'] ) ) {
+					throw new UserError( __( 'The "id" provided is invalid', 'wp-graphql-woocommerce' ) );
+				}
+				$order_id = absint( $id_components['id'] );
+			} elseif ( ! empty( $input['orderId'] ) ) {
+				$order_id = absint( $input['orderId'] );
+			} else {
+				throw new UserError( __( 'No order ID provided.', 'wp-graphql-woocommerce' ) );
+			}
+
+			Order_Mutation::add_order_meta( $order_id, $input, $context, $info );
+			Order_Mutation::add_items( $input, $order_id, $context, $info );
+
+			// Apply coupons.
+			if ( ! empty( $input['coupons'] ) ) {
+				Order_Mutation::apply_coupons( $order_id, $input['coupons'] );
+			}
+
+			$order = \WC_Order_Factory::get_order( $order_id );
+
+			// Make sure gateways are loaded so hooks from gateways fire on save/create.
+			WC()->payment_gateways();
+
+			// Validate customer ID.
+			if ( ! empty( $input['customerId'] ) ) {
```

Comment:
Suggestion: This one don't need to be nested. Can be used with a `&&`.

Reply:
Fixed in the checkout mutation branch